packages/web-api/src/controllers/scan-request-controller.ts

```diff
@@ -73,27 +74,38 @@ export class ScanRequestController extends ApiController {
 
         // tslint:disable-next-line: no-null-keyword
         this.logger.trackEvent('BatchScanRequestSubmitted', null, measurements);
+        let priority: string;
+        for (priority of Object.keys(processedData.requestCountByPriority)) {
+            this.logger.trackMetric('QueuedScanRequests', processedData.requestCountByPriority[priority], { priority: priority });
+        }
     }
 
     private getProcessedRequestData(batchId: string, scanRunRequests: ScanRunRequest[]): ProcessedBatchRequestData {
         const scanRequestsToBeStoredInDb: ScanRunBatchRequest[] = [];
         const scanResponses: ScanRunResponse[] = [];
+        const requestCountByPriority: { [priority: string]: number } = {};
 
         scanRunRequests.forEach(scanRunRequest => {
             const runRequestValidationResult = this.validateRunRequest(scanRunRequest);
             if (runRequestValidationResult.valid) {
                 // preserve GUID origin for a single batch scope
                 const scanId = this.guidGenerator.createGuidFromBaseGuid(batchId);
+                const priority = isNil(scanRunRequest.priority) ? 0 : scanRunRequest.priority;
                 scanRequestsToBeStoredInDb.push({
```

Comment:
If 0 is the default priority, I think we should extract a variable like ``DefaultPriority`` and save it at a proper place.

Reply:
Default is applied when property is not set. Priority is int with range -1000..1000. Just set int default